UI/WebServerResources/SOGoMailDataSource.js

```diff
@@ -82,9 +82,25 @@ SOGoMailDataSource = Class.create({
         },
 
         _loadCallback: function(http) {
+            var data, re;
             if (http.status == 200) {
                 if (http.responseText.length > 0) {
-                    var data = http.responseText.evalJSON(true);
+                    try {
+                        data = http.responseText.evalJSON(true);
+                    } catch(e) {
+                        if (e instanceof SyntaxError) {
+                            re = /action=["']\/SOGo\/connect["']/i;
+                            if (re.test(http.responseText)) {
+                                window.location= '/SOGo';
```

Comment:
This is wrong, what happens if I want to have sogo web under `http://my.mail.com/my/cool/webmail/` or `http://sogo.ohmycat.com`